manage_breast_screening/notifications/queries/failures.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+
+from django.db.models import F, QuerySet
+
+from manage_breast_screening.notifications.models import (
+    MessageStatus,
+    MessageStatusChoices,
+)
+
+
+class Failures:
+    def query(self, date: datetime = datetime.today()) -> QuerySet:
+        starts_at = date.replace(hour=0, minute=0, second=0)
+        ends_at = date.replace(hour=23, minute=59, second=59)
+
+        return (
+            MessageStatus.objects.filter(
+                message__appointment__starts_at__gte=starts_at,
+                message__appointment__starts_at__lte=ends_at,
+                status=MessageStatusChoices.FAILED.value,
```

Comment:
I think I'm a bit confused between the `MessageBatch` failures and the `Message` failures...
Currently a `Message` could be `failed` but its `MessageBatch` could be in a state of `failed_recoverable` in which case it is in process of retrying. 
In cases where the MessageBatch has unrecoverably failed to send, should the Messages be part of this failure report? Or will we triage those manually?
If we _don't_ want to include MessageBatch failures in this report, should we check the associated MessageBatch's status is `sent`?

Reply:
Oh I see that this is just querying `MessageStatuses` which are different to the `status` field on a Message, they are only updates from NHS Notify after the MessageBatch has been sent, and have the same choices as the `status` field...It's a bit confusing 😅 
So for MessageBatch failures we will be handling those through another support/triaging system?

Reply:
It does raise a very valid point about using `MessageStatusChoices` for both `Message#status` and `MessageStatus#status` model fields. I get the feeling we need to separate these.

Reply:
I wonder if there is even a need for `Message#status` - its status should only be determined either through the `MessageBatch#status` or its `MessageStatus` objects?